src/dir.cr

```diff
@@ -17,10 +15,7 @@ class Dir
 
   # Returns a new directory object for the named directory.
   def initialize(@path)
-    @dir = LibC.opendir(@path.check_no_null_byte)
-    unless @dir
-      raise Errno.new("Error opening directory '#{@path}'")
-    end
+    @dir = Crystal::System::Dir.each(@path)
```

Comment:
This ivar should perhaps be renamed to match the new meaning? For example `@entry_iterator`.

Reply:
Fixed